crates/db_views/community_follower/src/impls.rs

```diff
@@ -93,64 +116,48 @@ impl CommunityFollowerView {
 
   pub async fn list_approval_required(
     pool: &mut DbPool<'_>,
-    person_id: PersonId,
-    // TODO: if this is true dont check for community mod, but only check for local community
-    //       also need to check is_admin()
+    mod_id: PersonId,
     all_communities: bool,
-    pending_only: bool,
+    unread_only: bool,
     cursor_data: Option<CommunityActions>,
     page_back: Option<bool>,
     limit: Option<i64>,
   ) -> LemmyResult<Vec<PendingFollow>> {
     let conn = &mut get_conn(pool).await?;
     let limit = limit_fetch(limit)?;
-    let (person_alias, community_follower_alias) = diesel::alias!(
-      person as person_alias,
-      community_actions as community_follower_alias
-    );
 
-    // check if the community already has an accepted follower from the same instance
-    let is_new_instance = not(exists(
-      person_alias
-        .inner_join(
-          community_follower_alias.on(
-            person_alias
-              .field(person::id)
-              .eq(community_follower_alias.field(community_actions::person_id)),
-          ),
-        )
-        .filter(
-          person::instance_id
-            .eq(person_alias.field(person::instance_id))
-            .and(
-              community_follower_alias
-                .field(community_actions::community_id)
-                .eq(community_actions::community_id),
-            )
-            .and(
-              community_follower_alias
-                .field(community_actions::follow_state)
-                .eq(CommunityFollowerState::Accepted),
-            ),
-        ),
-    ));
+    // subquery to check if the community already has an accepted follower from the same instance
+    let ca_person_id = community_actions_instance_check.field(community_actions::person_id);
+    let ca_community_id = community_actions_instance_check.field(community_actions::community_id);
+    let ca_follow_state = community_actions_instance_check.field(community_actions::follow_state);
+    let p_person_id = person_instance_check.field(person::id);
+    let p_instance_id = person_instance_check.field(person::instance_id);
 
+    let is_new_instance = person_instance_check
+      .left_join(community_actions_instance_check.on(p_person_id.eq(ca_person_id)))
+      .into_boxed()
+      .filter(person::instance_id.eq(p_instance_id))
+      .filter(community::id.eq(ca_community_id))
+      .filter(ca_follow_state.eq(CommunityFollowerState::Accepted))
+      .select(1_i32.into_sql::<diesel::sql_types::Integer>());
```

Comment:
Had to convert this to a subquery to make it compile. Now it almost works, except the lines `.filter(person::instance_id` and `.filter(community::id` dont work. It seems the problem is that these tables dont appear in the from clause (only in the outer query). But the examples I found for subqueries work just like this ([example](https://stackoverflow.com/a/74300447)). 

Any idea how to fix it?

Reply:
I don't understand what's going on in this PR, because I don't know what you need.

You shouldn't need any moderator joining to the `CommunityFollowerView`. You just need to get their follow status right? In that case:

- add `community_actions` to `CommunityFollowerView`
- Don't filter by `followed_at` in L34, and use `community_actions.community_follower_state` to get their follow state.

Reply:
This is for moderators to get the count and list of [pending follows](https://github.com/LemmyNet/lemmy/blob/main/crates/api/api/src/community/pending_follows/list.rs) in a private community which need to be approved manually. For count it took a parameter community_id which doesnt make sense as it should automatically return the count for all private communities that the user moderates. For list it was already doing this, however the join was wrong so it actually checked if the user who wants to follow the private community is a moderator of that community. So I had to add a separate join for the mod.

The logic is correct now as shown by the test, and it actually passes with L139 and L140 commented out. But of course these lines are required.

Reply:
Okay made those changes. Now theres still the same problem as before, the subquery for `is_new_instance` doesnt compile if the filters for `instance_id` and `community_id` are enabled.

https://github.com/LemmyNet/lemmy/pull/6003/files#diff-3923c280a45c47ea8bc986dcbeb831efc1c23cddd175899d9c7cd72076740715R81